cli/flox-watchdog/src/main.rs

```diff
@@ -151,8 +151,6 @@ fn run(args: Cli) -> Result<(), Error> {
     let activation = ActivationPid::from(args.pid);
     // Check if our original parent has already exited.
     // If it has, we can go ahead and cleanup.
-    // Note that this check should come after we call prctl or kqueue so that
-    // there isn't a race between checking ppid and calling prctl or kqueue.
```

Comment:
The race is
- Check if parent is alive
- Parent exits so you get reparented
- Call prctl

So instead we did:
- Call prctl
- Make sure the parent you're waiting for is your original parent

I think we can drop the entire code block now that we aren't using `prctl`. kqueue will fail if the PID has already exited

Reply:
Because we would never get a termination event (SIGUSR1) if the watchdog was re-parented before calling `prctl`? I think I was thrown by the two mentions of `kqueue`.

Agree that we shouldn't need the code block now that we're:
1. Passing the PID in rather than calculating it from current parent
2. Always receiving termination events (which will conditionally clean-up) if that PID exits

I can follow-up in a separate PR.

Reply:
Yeah I think at the time I wrote the bit about kqueue I didn't realize kqueue would error if the parent had exited. We now handle that since https://github.com/flox/flox/pull/2018